lib/parameter_substitution/expression.rb

```diff
@@ -98,5 +134,14 @@ def unknown_parameters
           expression.unknown_parameters(inside_quotes)
         end.compact
     end
+
+    def unknown_parameter_methods
+      @expression_list.select(&:parameter_name).reduce({}) do |hash, expression|
+        hash[expression.parameter_name] ||= []
+        hash[expression.parameter_name]
+          .push(*(methods_used_by_expression(expression) - ParameterSubstitution::Formatters::Manager.all_formats.map { |k, _v| k.to_s }))
+        hash
+      end
+    end
```

Comment:
Can we simplify this method and maybe extract some well-named variables or methods?

Reply:
I extracted out a method that hopefully simplifies things a bit. Let me know what you think or if have an idea for a different approach.

Reply:
It's pushed up.

Reply:
Thanks, Juan. This reads better to me now.